Processes.md

```diff
@@ -21,42 +21,114 @@ Up to date
-added to the menu. The function responsible for that is shown below;
+start_link begins a spawn linked to the current process with the given functions, the pid is registered using `Process.register`. For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own process then passed the initial state which in our case will change from an empty [] to an actual list of items to be added to the menu. The function responsible for that is shown below;
 
-**add code**
+ ```
+defmodule Menuprocess do
+  
+  #menu server
+  @doc"""
+  This function initializes our server 
+  """
+   @spec start_link(any()) :: pid()
+    def start_link(initial_state \\ []) do
+     pid = spawn(__MODULE__, :menu_state, [initial_state])
+     Process.register(pid, :process)
+     pid 
+    end
+end 
+```
 
 Return the current menu state:
 To return the current state of the menu, we basically need to send a message from the client requesting for all items
 present in our server to our mailbox which is handled by the receive do (whose work is to await requests and match them 
 to the messages in the mailbox), then the server responds with the state of the menu.
 
-**add code**
+```
+#server
+
+@spec menu_state(tuple()) :: {:ok, list(String.t())}
+  def menu_state(state) do
+    receive do
+      {caller, :items} ->
+        send caller, state
+        menu_state(state)
+    end
+  end
+  
+#from client
+
+@spec all() :: {:ok, list(String.t())}
+    def all do 
+      send :process, {self(), :items}
+        receive do state -> inspect state end
+    end
+```
 
 Adding a list of items to our menu list is handled by a different function which takes in {self(), :additems, items}
 (from the client) the same way we return the menu state is the same way we add items to our menu, the receive do
 function takes the message sent then looks up the mailbox and matches it to {caller, :additems, item} which in turn
 adds the items to the menu and returns the new_state of the menu.
 
+```
+#server
+
+@spec menu_state(tuple()) :: {:ok, list(String.t())}
+  def menu_state(state) do
+    receive do
+     {caller, :additems, item} ->
+        new_state = [item | state]
+        send caller, length new_state
+        menu_state(new_state)
+    end
+  end
+        
+#from client
 
-**add code**
+ @spec add_item(String.t()) :: {:ok, list(String.t())}
+    def add_item(item)do
+      send :process, {self(), :additems, item}
+        receive do response -> inspect response end
+    end
+```
 
 To handle all wrong entries that don’t match any messages in the mailbox, we use the function below.
 
-**add code**
+```
+#server
+ @spec menu_state(tuple()) :: {:ok, list(String.t())}
+  def menu_state(state) do
+    receive do
+     _error ->
+       IO.puts "Wrong entry"
+       menu_state(state)
+    end
+  end
+```
 
-**add iex code**
+Fire up iex and excute the following commands.
```

Comment:
I would sprinkle this demos around the articles as their code comes up rather than  bunch them all up at the end